store/pruning/manager.go

```diff
@@ -0,0 +1,137 @@
+package pruning
+
+import (
+	"sync"
+
+	"cosmossdk.io/log"
+	"cosmossdk.io/store/v2"
+)
+
+// Manager is an abstraction to handle the pruning logic.
+type Manager struct {
+	stateStore      store.VersionedDatabase
+	stateCommitment store.Committer
+
+	logger         log.Logger
+	storeOpts      Options
+	commitmentOpts Options
+
+	chStore      chan struct{}
+	chCommitment chan struct{}
+
+	mtx sync.Mutex
+}
+
+// NewManager creates a new Manager instance.
+func NewManager(
+	stateStore store.VersionedDatabase,
+	stateCommitment store.Committer,
+	logger log.Logger,
+) *Manager {
```

Comment:
```suggestion
func NewManager(logger log.Logger, ss store.VersionedDatabase, sc store.Committer) *Manager {
```

Reply:
should we pass in opts too?

Reply:
just following same patterns of iavl options